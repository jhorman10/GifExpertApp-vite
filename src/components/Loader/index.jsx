export const Loader = () => {
  return (
    <div className="lds-facebook" title="loader">
      <div></div>
      <div></div>
      <div></div>
    </div>
  );
};
