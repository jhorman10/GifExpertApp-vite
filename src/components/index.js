//archivo indice o de barril
export * from './AddCategory';
export * from './GifGrid';
export * from './GifGridItem';
export * from './Loader';
