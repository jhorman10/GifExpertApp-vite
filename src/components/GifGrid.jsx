import { useFetchGifs } from '../hooks/useFetchGifs';
import { GifGridItem } from './GifGridItem';
import { Loader } from './Loader';

export const GifGrid = ({ category }) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <div className='main-container'>
      <h3>{category}</h3>
      {isLoading && <Loader />}
      <div className="card-grid">
        {images.map((image) => (
          <GifGridItem key={image.id} {...image} />
        ))}
      </div>
    </div>
  );
};
