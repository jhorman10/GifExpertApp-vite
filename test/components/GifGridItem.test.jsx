import { render, screen } from '@testing-library/react';
import { GifGridItem } from '../../src/components/GifGridItem';

describe('Pruebas en <GifGridItem />', () => {
  const title = 'Saitama';
  const url = 'https://media1.giphy.com/media/l0Hl6qQ7zQQQQQQQ/giphy.gif';

  test('Debe hacer match con el snapshot', () => {
    const { container } = render(<GifGridItem title={title} url={url} />);
    expect(container).toMatchSnapshot();
  });

  test('Debe mostrar la imagen con el URl y el ALT indicado', () => {
    render(<GifGridItem title={title} url={url} />);
    const { alt, src } = screen.getByRole('img');
    expect(alt).toBe(title);
    expect(src).toBe(url);
  });

  test('Debe mostrar el tÃ­tulo en el componente', () => {
    render(<GifGridItem title={title} url={url} />);
    expect(screen.getByText(title)).toBeTruthy();
  });
});
