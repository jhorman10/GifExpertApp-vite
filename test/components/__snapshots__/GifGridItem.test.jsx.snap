// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <GifGridItem /> Debe hacer match con el snapshot 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Saitama"
      src="https://media1.giphy.com/media/l0Hl6qQ7zQQQQQQQ/giphy.gif"
    />
    <p>
      Saitama
    </p>
  </div>
</div>
`;
